mqtt/mqtt-bridge/src/client.rs

```diff
@@ -300,30 +300,41 @@ impl<T: EventHandler> MqttClient<T> {
         }
     }
 
-    pub fn shutdown_handle(&self) -> Result<ShutdownHandle, ShutdownError> {
+    pub fn shutdown_handle(&self) -> Result<ClientShutdownHandle, ShutdownError> {
         self.client
             .shutdown_handle()
             .map_or(Err(ShutdownError::ClientDoesNotExist), |shutdown_handle| {
-                Ok(ShutdownHandle(shutdown_handle))
+                Ok(ClientShutdownHandle(shutdown_handle))
             })
     }
 
-    pub async fn handle_events(mut self) -> Result<(), ClientConnectError> {
+    pub fn publish_handle(&self) -> Result<PublishHandle, ClientError> {
+        let publish_handle = self
+            .client
+            .publish_handle()
+            .map_err(ClientError::PublishHandle)?;
+
+        Ok(publish_handle)
+    }
+
+    pub async fn handle_events(&mut self) {
+        // TODO REVIEW: Is client id ever not set
+        let client_id = self.client_id.clone().unwrap_or(String::new());
```

Comment:
Client id is defined in this client wrapper as an option. Will it ever be set to None (i.e. clean session)?